File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/auth/payload/request/VerificationEmailRequest.java
package org.bootstrapbugz.api.auth.payload.request;

import jakarta.validation.constraints.NotBlank;
import org.bootstrapbugz.api.shared.validator.UsernameOrEmail;

public record VerificationEmailRequest(
    @NotBlank(message = "{usernameOrEmail.required}")
        @UsernameOrEmail(message = "{usernameOrEmail.invalid}")
        String usernameOrEmail) {}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/payload/request/VerificationEmailRequest.kt
package org.bootstrapbugz.api.auth.payload.request

import jakarta.validation.constraints.NotBlank
import org.bootstrapbugz.api.shared.validator.UsernameOrEmail

data class VerificationEmailRequest(
    @UsernameOrEmail(message = "{usernameOrEmail.invalid}")
    @NotBlank(message = "{usernameOrEmail.required}")
    val usernameOrEmail: String
)
